pkg/server/scrub.go

```diff
@@ -0,0 +1,35 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// scrubbing utilities to remove user information from arbitrary objects
+
+const ScrubbedReplacement = "<scrubbed>"
+
+// Scrub returns a fmt.Formatter to be used with fmt.Printf and related functions that prints a
+// a copy of i with the fields containing user data (ie, Environment) set to `"<scrubbed>"`.
+// Objects passed in should as copies of the original, and *NOT* be pointers
+func (c *criService) Scrub(i interface{}) fmt.Formatter {
+	if !c.config.ScrubLogs {
+		return spew.NewFormatter(i)
+	}
+
+	switch o := i.(type) {
```

Comment:
I know it's already discussed some in https://github.com/microsoft/hcsshim/pull/1315 but I think we should really consider looking at a more generic route. My concerns are that this isn't comprehensive, and this doesn't help to prevent us or inform us from logging sensitive information in the future.

Reply:
I think so to, I wanted a quick solution here, but I do want to explore how to make it more robust.
Unfortunately, the spec and container config structs are defined in different, external repos, so we would have to duplicate them locally if we wanted to add field annotations. Otherwise, scanning/parsing all logs via a logrus hook seemed too expensive.

Reply:
I don't think we'd want to duplicate them. I think Justin's suggestion about working with upstream to make that happen would be great. I'm okay with taking this change so long as we take an item to keep looking into how to improve it and work with upstream.

Reply:
agreed; having to remember to add `Scrub(` calls around logging will definitely fail and cause problems